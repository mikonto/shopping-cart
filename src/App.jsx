import GlobalStyle from "./GlobalStyle";
import Header from "./Header";

function App() {
  return (
    <>
      <GlobalStyle />
      <Header />
    </>
  );
}

export default App;
