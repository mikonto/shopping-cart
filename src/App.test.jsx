import { render, screen } from "@testing-library/react";
import App from "./App";
import { MemoryRouter, Route, Routes } from "react-router-dom";

describe("App Component", () => {
  it("renders the Header component", () => {
    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );

    const logoImage = screen.getByRole("img");
    expect(logoImage).toBeInTheDocument();
    expect(logoImage.src).toMatch(/logo\.png$/);
  });

  it("renders the correct links", () => {
    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
    const homeLink = screen.getByRole("link", { name: "Home" });
    const productsLink = screen.getByRole("link", { name: "Products" });
    const shoppingCartLink = screen.getByTestId("shopping-cart-link");

    expect(homeLink).toBeInTheDocument();
    expect(productsLink).toBeInTheDocument();
    expect(shoppingCartLink).toBeInTheDocument();

    expect(homeLink).toHaveAttribute("href", "/");
    expect(productsLink).toHaveAttribute("href", "/products");
    expect(shoppingCartLink).toHaveAttribute("href", "/shopping-cart");
  });

  it("renders the Home component for the default route", () => {
    render(
      <MemoryRouter initialEntries={["/"]}>
        <App />
      </MemoryRouter>
    );
    expect(screen.getByText("Welcome to Webshop!")).toBeInTheDocument();
  });

  it("renders the Products component for /products route", () => {
    render(
      <MemoryRouter initialEntries={["/products"]}>
        <App />
      </MemoryRouter>
    );
    expect(screen.getByText("Products")).toBeInTheDocument();
  });

  it("renders the ShoppingCart component for /shopping-cart route", () => {
    render(
      <MemoryRouter initialEntries={["/shopping-cart"]}>
        <Routes>
          <Route path="/:name" element={<App />} />
        </Routes>
      </MemoryRouter>
    );

    expect(screen.getByText("Shopping cart")).toBeInTheDocument();
  });

  it("it renders correctly", () => {
    const { container } = render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
