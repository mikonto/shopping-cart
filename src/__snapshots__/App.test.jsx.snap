// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Component > it renders correctly 1`] = `
<div>
  <div
    class="sc-bYMqJM jWygDP"
  >
    <div
      style="position: fixed; z-index: 9999; top: 10px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
    />
    <div
      class="sc-gswONk hXrFBL"
    >
      <a
        href="/"
      >
        <img
          alt="Description of the image"
          class="sc-bcXGCL fuelPu"
          src="/src/assets/logo.png"
        />
      </a>
      <nav
        class="sc-dkrGBB euNLZT"
      >
        <a
          class="sc-hLBatu dWMTYc"
          href="/"
        >
          Home
        </a>
        <a
          class="sc-hLBatu dWMTYc"
          href="/products"
        >
          Products
        </a>
        <a
          class="sc-hLBatu dWMTYc"
          data-testid="shopping-cart-link"
          href="/shopping-cart"
        >
          <span
            class="MuiBadge-root css-1c32n2y-MuiBadge-root"
            data-testid="badge"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ShoppingCartIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
            <span
              class="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorPrimary css-y2h6q0-MuiBadge-badge"
            />
          </span>
        </a>
      </nav>
    </div>
    <div
      class="sc-eDvShL kTAuDa"
    >
      <div
        class="sc-jSVask cmDsPA"
      >
        <h1
          class="sc-gKPSgB bBOYpf"
        >
          Welcome to Webshop!
        </h1>
      </div>
      <div
        class="sc-jSVask cmDsPA"
      >
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Cras pulvinar mattis nunc sed.
        </p>
      </div>
      <div
        class="sc-jSVask cmDsPA"
      >
        <a
          href="/products"
        >
          <button
            class="sc-iBYPxa gyjTni"
          >
            Shop now
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
